opm/simulators/flow/Main.cpp

```diff
@@ -0,0 +1,236 @@
+            if (err != DAMARIS_OK) {
+                std::cerr << "ERROR: Damaris library produced an error result for damaris_stop()" << std::endl;
+            }
+        }
+        err = damaris_finalize();
+        if (err != DAMARIS_OK) {
+            std::cerr << "ERROR: Damaris library produced an error result for damaris_finalize()" << std::endl;
+        }
+    }
+#endif // HAVE_DAMARIS
+
+#if HAVE_MPI && !HAVE_DUNE_FEM
+    MPI_Finalize();
+#endif
+}
+
+void Main::setArgvArgc_(const std::string& filename)
+{
+    this->deckFilename_ = filename;
+    this->flowProgName_ = "flow";
+
+    this->argc_ = 2;
+    this->saveArgs_[0] = const_cast<char *>(this->flowProgName_.c_str());
+    this->saveArgs_[1] = const_cast<char *>(this->deckFilename_.c_str());
+
+    // Note: argv[argc] must exist and be nullptr
+    assert ((sizeof this->saveArgs_) > (this->argc_ * sizeof this->saveArgs_[0]));
+    this->saveArgs_[this->argc_] = nullptr;
+
+    this->argv_ = this->saveArgs_;
+}
+
+void Main::initMPI()
+{
+#if HAVE_DUNE_FEM
+    Dune::Fem::MPIManager::initialize(argc_, argv_);
+#elif HAVE_MPI
+    MPI_Init(&argc_, &argv_);
+#endif
+    EclGenericVanguard::setCommunication(std::make_unique<Parallel::Communication>());
+
+    handleTestSplitCommunicatorCmdLine_();
+
+#if HAVE_MPI
+    if (test_split_comm_ && EclGenericVanguard::comm().size() > 1) {
+        int world_rank = EclGenericVanguard::comm().rank();
+        int color = (world_rank == 0);
+        MPI_Comm new_comm;
+        MPI_Comm_split(EclGenericVanguard::comm(), color, world_rank, &new_comm);
+        isSimulationRank_ = (world_rank > 0);
+        EclGenericVanguard::setCommunication(std::make_unique<Parallel::Communication>(new_comm));
+    }
+#endif // HAVE_MPI
+}
+
+void Main::handleVersionCmdLine_(int argc, char** argv)
+{
+    auto pos = std::find_if(argv, argv + argc,
+        [](const char* arg)
+    {
+        return std::strcmp(arg, "--version") == 0;
+    });
+
+    if (pos != argv + argc) {
+        std::cout << "flow " << moduleVersionName() << std::endl;
+        std::exit(EXIT_SUCCESS);
+    }
+}
+
+void Main::handleTestSplitCommunicatorCmdLine_()
+{
+    if (argc_ >= 2 && std::strcmp(argv_[1], "--test-split-communicator=true") == 0) {
+        test_split_comm_ = true;
+        --argc_;             // We have one less argument.
+        argv_[1] = argv_[0]; // What used to be the first proper argument now becomes the command argument.
+        ++argv_;             // Pretend this is what it always was.
+    }
+}
+
+void Main::readDeck(const std::string& deckFilename,
+                    const std::string& outputDir,
+                    const std::string& outputMode,
+                    const bool init_from_restart_file,
+                    const bool allRanksDbgPrtLog,
+                    const bool strictParsing,
+                    const int mpiRank,
+                    const int output_param)
+{
+    auto python = std::make_shared<Python>();
```

Comment:
If you're changing this anyway, you might as well move the `make_shared<Python>()` call to the argument list of `Opm::readDeck()`.  That way we don't incur the extra shared-pointer copy when calling `readDeck()`.  Alternatively, you might use `std::move(python)` in the argument list.

Reply:
moved the make_shared call.